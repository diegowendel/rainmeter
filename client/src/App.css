.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

.clicable {
  cursor: pointer;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.forecast-panel {
  background-color: #fff;
  border-radius: 5px;
  border: 1px solid #ddd;
  display: flex;
  flex-direction: column;
  padding: 15px;
}

.forecast-panel-details {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
}

.forecast-panel-details-info {
  color: #878787;
  text-align: left;
}

.forecast-panel-details-right {
  display: flex;
  flex-direction: column;
  align-items: baseline;
  justify-content: flex-start;
  color: #878787;
  text-align: left;
}

.forecast-temperature-highlight {
  color: #212121;
  font-size: 64px;
}

.flex-column {
  display: flex;
  flex-direction: column;
}

.flex-row {
  display: flex;
  flex-direction: row;
}

.forecast-cards {
  background-color: #fff;
  border-radius: 10px;
  border: 1px solid #ddd;
  display: flex;
  flex-direction: row;
  padding: 15px;
}

.forecast-card {
  border: 1px solid transparent;
  border-radius: 3px;
  display: flex;
  flex: 1;
  flex-direction: column;
  justify-content: space-between;
}

.forecast-temperature {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  color: #bababa;
}

.forecast-rain {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-content: center;
}

.forecast-wind {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-content: center;
}

.forecast-weekday {
  align-self: center;
  color: #bababa;
}

.forecast-selected-day {
  background-color: #fcfcfc;
  border: 1px solid #e9e9e9;
  border-radius: 3px;
  margin: 0;
}
