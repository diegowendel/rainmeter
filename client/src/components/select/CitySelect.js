import React from 'react';
import Select from 'react-select';
import { FormattedMessage } from 'react-intl';

const CitySelect = (props) => {
  return (
    <Select className="basic-single mb-1 w-600"
      classNamePrefix="select"
      isDisabled={props.isDisabled}
      isLoading={props.isLoading}
      isClearable={true}
      isRtl={false}
      isSearchable={true}
      getOptionLabel={(option) => option.name}
      getOptionValue={(option) => option.id}
      options={props.options}
      onChange={(selectedCity) => props.onChange("selectedCity", selectedCity)}
      value={props.value}
      placeholder={<FormattedMessage id="CitySelect.placeholder" />} />
  );
};

export default CitySelect;
