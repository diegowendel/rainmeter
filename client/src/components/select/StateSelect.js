import React from 'react';
import Select from 'react-select';
import { stateOptions } from '../../data/Data';

const StateSelect = (props) => {
  return (
    <Select className="basic-single"
      classNamePrefix="select"
      isDisabled={false}
      isLoading={false}
      isClearable={true}
      isRtl={false}
      isSearchable={true}
      options={stateOptions}
      onChange={(selectedState) => props.onChange("selectedState", selectedState)}
      value={props.value}
      placeholder="Selecione seu estado..." />
  );
};

export default StateSelect;
