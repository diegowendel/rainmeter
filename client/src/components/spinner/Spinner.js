import React from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faSpinner } from '@fortawesome/free-solid-svg-icons/index';
import '../../styles/Spinner.scss';

function Spinner() {
  return (
    <div className="spinner">
      <FontAwesomeIcon className="spinnerIcon" icon={faSpinner} size="2x" spin />
    </div>
  );
}

export default Spinner;
