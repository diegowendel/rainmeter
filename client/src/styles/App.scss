.clicable {
  cursor: pointer;
}

.container-center {
  @extend .flex-column;
  align-items: center;
}

.d-none {
  display: none;
}

.flex-column {
  display: flex;
  flex-direction: column;
}

.flex-row {
  display: flex;
  flex-direction: row;
}

.link-climatempo {
  color: #878787;
  display: block;
  font-size: 11px;
  margin: 5px;
  padding-left: 20px;
  text-decoration: none;
}

.link-climatempo:hover {
  color: #878787;
}

.w-600 {
  width: 600px;
}
