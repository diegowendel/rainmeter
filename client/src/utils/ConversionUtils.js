class ConversionUtils {

  /**
   * @param temperature in celsius
   */
  static toFahrenheit(temperature) {
    return Math.round(temperature * 1.8 + 32);
  }

  /**
   * @param speed in km/h
   */
  static toMPH(speed) {
    return Math.round(speed * 0.6214);
  }
}

export default ConversionUtils;
