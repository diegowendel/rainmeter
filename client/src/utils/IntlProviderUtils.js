import React, { cloneElement } from 'react';
import { IntlProvider } from 'react-intl';
import translations from '../locale/translations.json';

export const intl = (component, locale) => {
  const language = locale || "en-US";
  return (
    <IntlProvider locale={language}
      messages={translations[language]}>
        { cloneElement(component) }
    </IntlProvider>
  );
};
