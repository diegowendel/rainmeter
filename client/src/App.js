import React, { Component, Fragment } from 'react';
import './App.css';
import API from './service/ClimaTempoAPI';
import { CitySelect, StateSelect } from './components/select';
import ForecastPanel from './components/forecast/ForecastPanel';

class App extends Component {

  constructor(props) {
    super(props);

    this.state = {
      cities: undefined,
      forecast: undefined,
      loadingCities: undefined,
      selectedCity: undefined,
      selectedState: undefined,
      text: undefined
    };

    this.findCities = this.findCities.bind(this);
    this.onChange = this.onChange.bind(this);
    this.onPress = this.onPress.bind(this);
  }

  findCities(state) {
    this.setState({loadingCities: true});
    API.getCitiesFromState(state).then(res => {
      this.setState({
        cities : res.data,
        loadingCities: false
      });
    }).catch(/*todo*/);
  }

  onChange(propertyName, property) {
    if(property && propertyName === "selectedState") {
      this.findCities(property.value);
    }

    this.setState({ [propertyName]: property });
  }

  onPress() {
    API.getForecastSevenDays(this.state.selectedCity.id).then(res => {
      console.log('WEATHER', res);
      this.setState({forecast: res.data});
    }).catch(/* todo */);
  }

  render() {
    return (
      <div className="container">
        <header className="">

          <StateSelect onChange={this.onChange}
            value={this.state.selectedState} />

          <CitySelect
            isDisabled={!this.state.selectedState}
            isLoading={this.state.loadingCities}
            onChange={this.onChange}
            options={this.state.cities}
            value={this.state.selectedCity} />

          {this.state.forecast &&
            <ForecastPanel
              forecast={this.state.forecast}
            />
          }

          <button onClick={this.onPress}>Click Me!</button>
        </header>
      </div>
    )
  }
}

export default App;
