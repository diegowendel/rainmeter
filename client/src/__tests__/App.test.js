import React from 'react';
import { intl } from '../utils/IntlProviderUtils';
import { mount } from 'enzyme';

import App from '../App';
import Spinner from '../components/spinner/Spinner';
import { StateSelect } from '../components/select';

describe("<App />", () => {
  it("Should show spinner when loading", () => {
    const onChangeLocale = jest.fn();
    const wrapper = mount(intl(<App locale="pt-BR" onChangeLocale={onChangeLocale} />, "pt-BR"));
    const appInstance = wrapper.find(App).instance();

    expect(appInstance.props.locale).toBe("pt-BR");

    expect(wrapper.find(Spinner)).toHaveLength(0);

    appInstance.setState({ loading: true }, () => {
      wrapper.update(); // update must be called
      expect(wrapper.find(Spinner)).toHaveLength(1);
    });
  });

  it("Renders correctly change in state", () => {
    const onChangeLocale = jest.fn();
    const wrapper = mount(intl(<App locale="pt-BR" onChangeLocale={onChangeLocale} />, "pt-BR"));
    const appInstance = wrapper.find(App).instance();

    expect(appInstance.props.locale).toBe("pt-BR");

    // Find input field on Select component (from the react-select module)
    const input = wrapper.find(StateSelect).find("input");

    // Simulate the arrow down event to open the dropdown menu.
    input.simulate("keyDown", { key: "ArrowDown", keyCode: 40 });

    // Simulate the enter key to select the first option.
    input.simulate("keyDown", { key: "Enter", keyCode: 13 });

    expect(appInstance.state.selectedState.label).toBe("Acre");
  });
});
