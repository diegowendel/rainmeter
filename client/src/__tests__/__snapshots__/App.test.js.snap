// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
<div>
  <div
    class="container container-center"
  >
    <div>
      <div
        class="css-1pcexqc-container basic-single mb-1 mt-3 w-600"
      >
        <div
          class="css-bg1rzq-control select__control"
        >
          <div
            class="css-1hwfws3 select__value-container"
          >
            <div
              class="css-151xaom-placeholder select__placeholder"
            >
              Selecione seu estado...
            </div>
            <div
              class="css-1g6gooi"
            >
              <div
                class="select__input"
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-2-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class="css-1wy0on6 select__indicators"
          >
            <span
              class="css-bgvzuu-indicatorSeparator select__indicator-separator"
            />
            <div
              aria-hidden="true"
              class="css-16pqwjk-indicatorContainer select__indicator select__dropdown-indicator"
            >
              <svg
                aria-hidden="true"
                class="css-19bqh2r"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-ssi0ig-container basic-single mb-1 w-600 select--is-disabled"
      >
        <div
          class="css-14ojxsb-control select__control select__control--is-disabled"
        >
          <div
            class="css-1hwfws3 select__value-container"
          >
            <div
              class="css-151xaom-placeholder select__placeholder"
            >
              Selecione sua cidade...
            </div>
            <div
              class="css-1g6gooi"
            >
              <div
                class="select__input"
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  disabled=""
                  id="react-select-3-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class="css-1wy0on6 select__indicators"
          >
            <span
              class="css-622gbt-indicatorSeparator select__indicator-separator"
            />
            <div
              aria-hidden="true"
              class="css-16pqwjk-indicatorContainer select__indicator select__dropdown-indicator"
            >
              <svg
                aria-hidden="true"
                class="css-19bqh2r"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="landing-panel"
      >
        <span>
          Busque pela sua cidade e veja se vai precisar passar protetor solar.
        </span>
        <img
          alt="sol"
          class="icon"
          src="/img/1.svg"
        />
        <span>
          Se vai precisar de um guarda-chuva.
        </span>
        <img
          alt="chuva"
          class="icon"
          src="/img/6.svg"
        />
        <span>
          Ou se Ã© melhor levar um casaco.
        </span>
        <img
          alt="neve"
          class="icon"
          src="/img/7.svg"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`renders without crashing 1`] = `
<div>
  <div
    class="container container-center"
  >
    <div>
      <div
        class="css-1pcexqc-container basic-single mb-1 mt-3 w-600"
      >
        <div
          class="css-bg1rzq-control select__control"
        >
          <div
            class="css-1hwfws3 select__value-container"
          >
            <div
              class="css-151xaom-placeholder select__placeholder"
            >
              Selecione seu estado...
            </div>
            <div
              class="css-1g6gooi"
            >
              <div
                class="select__input"
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  id="react-select-2-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class="css-1wy0on6 select__indicators"
          >
            <span
              class="css-bgvzuu-indicatorSeparator select__indicator-separator"
            />
            <div
              aria-hidden="true"
              class="css-16pqwjk-indicatorContainer select__indicator select__dropdown-indicator"
            >
              <svg
                aria-hidden="true"
                class="css-19bqh2r"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="css-ssi0ig-container basic-single mb-1 w-600 select--is-disabled"
      >
        <div
          class="css-14ojxsb-control select__control select__control--is-disabled"
        >
          <div
            class="css-1hwfws3 select__value-container"
          >
            <div
              class="css-151xaom-placeholder select__placeholder"
            >
              Selecione sua cidade...
            </div>
            <div
              class="css-1g6gooi"
            >
              <div
                class="select__input"
                style="display: inline-block;"
              >
                <input
                  aria-autocomplete="list"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  disabled=""
                  id="react-select-3-input"
                  spellcheck="false"
                  style="box-sizing: content-box; width: 2px; border: 0px; font-size: inherit; opacity: 1; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  style="position: absolute; top: 0px; left: 0px; visibility: hidden; height: 0px; overflow: scroll; white-space: pre; font-size: inherit; font-family: -webkit-small-control; letter-spacing: normal; text-transform: none;"
                />
              </div>
            </div>
          </div>
          <div
            class="css-1wy0on6 select__indicators"
          >
            <span
              class="css-622gbt-indicatorSeparator select__indicator-separator"
            />
            <div
              aria-hidden="true"
              class="css-16pqwjk-indicatorContainer select__indicator select__dropdown-indicator"
            >
              <svg
                aria-hidden="true"
                class="css-19bqh2r"
                focusable="false"
                height="20"
                viewBox="0 0 20 20"
                width="20"
              >
                <path
                  d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                />
              </svg>
            </div>
          </div>
        </div>
      </div>
      <div
        class="landing-panel"
      >
        <span>
          Busque pela sua cidade e veja se vai precisar passar protetor solar.
        </span>
        <img
          alt="sol"
          class="icon"
          src="/img/1.svg"
        />
        <span>
          Se vai precisar de um guarda-chuva.
        </span>
        <img
          alt="chuva"
          class="icon"
          src="/img/6.svg"
        />
        <span>
          Ou se Ã© melhor levar um casaco.
        </span>
        <img
          alt="neve"
          class="icon"
          src="/img/7.svg"
        />
      </div>
    </div>
  </div>
</div>
`;
